### PR TITLE
Configure clean pod timeout during runtime

### PR DESCRIPTION
**Pull request by dpaks**
Provided a way to specify timeout for pod shutdown in pool's custom config.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ const (
 	COMPOSE_HTTP_TIMEOUT                 = "launchtask.composehttptimeout"
 	HTTP_TIMEOUT                         = "launchtask.httptimeout"
 	COMPOSE_STOP_TIMEOUT                 = "cleanpod.timeout"
+	DEFAULT_COMPOSE_STOP_TIMEOUT         = 20
```
Comment (kmala) [inline config/config.go]: why is this changed from 10?

File: config/config.go [hunk 2]
```diff
@@ -173,8 +174,31 @@ func GetLaunchTimeout() time.Duration {
 	return duration
 }
 
+// GetStopTimeout returns the grace period time for a pod to die
+// Returns time in seconds as an integer
 func GetStopTimeout() int {
-	return GetConfig().GetInt(COMPOSE_STOP_TIMEOUT)
+	// value from the config file expressed as seconds in integer format
+	// Viper implicitly casts string to int too
+	if valInt := GetConfig().GetInt(COMPOSE_STOP_TIMEOUT); valInt != 0 {
```
Comment (kmala) [inline config/config.go]: Can't you change the default also as string so that you need not do so many validations

File: config/config.go (hunk 1 above)
Comment (dpaks) [reply config/config.go]: The dce readme and config file says default 20. Hence changed. However, the default docker stop timeout is 10. Should I revert?

File: config/config.go (hunk 2 above)
Comment (dpaks) [reply config/config.go]: Regardless of the default, we can take this as a string. However, if I do a GetString(), I'll need a strconv.Atoi() as well. Should I do similar to GetLaunchTimeout() ?

Comment (kmala) [reply config/config.go]: what i meant was set the default as `10s` and you can just get using config.getDuration() right?

File: config/config.go (hunk 1 above)
Comment (kmala) [reply config/config.go]: i will let @JiaminZhu decide then

File: config/config.go (hunk 2 above)
Comment (dpaks) [reply config/config.go]: In that case, https://github.com/paypal/dce-go/blob/95b7badfcd238958dbbdc99c84950bde1d2bf3b5/config/config.yaml#L20 will have to be changed to 20s. Also, the config under all the plugins need to be changed too.

Comment (dpaks) [reply config/config.go]: @kmala @JiaminZhu Should I set the default for clean pod timeout as duration everywhere? E.g: '20s' in stead of '20'

Comment (kmala) [reply config/config.go]: my suggestion is to go with duration everywhere so its clean and simple

File: config/config.go (hunk 1 above)
Comment (JiaminZhu) [reply config/config.go]: timeout can be changed through config file, but let's keep the default as 10

Comment (dpaks) [reply config/config.go]: Done, reverted to 10

Comment (kmala) [reply config/config.go]: i still see it as 20 and it should be `10s`

File: config/config.go [hunk 3]
```diff
@@ -173,8 +174,25 @@ func GetLaunchTimeout() time.Duration {
 	return duration
 }
 
+// GetStopTimeout returns the grace period time for a pod to die
+// Returns time in seconds as an integer
 func GetStopTimeout() int {
-	return GetConfig().GetInt(COMPOSE_STOP_TIMEOUT)
+	// the timeout value is expressed as duration in string format
+	valStr := GetConfig().GetString(COMPOSE_STOP_TIMEOUT)
```
Comment (kmala) [inline config/config.go]: we can use use directly getDuration right since viper supports it?

Comment (dpaks) [reply config/config.go]: GetDuration() internally casts various data types. For example, it accepts both 25 and "25", but outputs 25ns. 
Since we now supports only duration format for timeout, the error handling (getduration returns no error, it is ignored iternally) based on zero value will be tricky. Error handling is required to return default timeout. @kmala what do you think?

Comment (ridv) [reply config/config.go]: The value will have to be validated somewhere. It's probably easier in this case to use `GetConfig().GetDuration(key)` because it let's you offload the parsing then you can validate the value later.

For example:

"0m" and "0s" would both evaluate to a "0s" duration, which you can validate afterwards by checking that the duration is 0.

Comment (dpaks) [reply config/config.go]: @ridv If the input is specified with unit of time ("25s"), then we're good and requires no more validation.
As part of this jira, we have changed the format of this timeout from integer to duration type (25 to 25s). We now disallow timeout to be specified as 25 or "25" (25 is just an example). GetDuration accepts both 25 and "25" and treats them as value in nanoseconds and conversion to seconds gives 0 as the value. If the value is coming as 0 seconds, we again need to return the default. Let me make this change.

File: config/config.go [hunk 4]
```diff
@@ -58,6 +58,7 @@ const (
 	COMPOSE_HTTP_TIMEOUT                 = "launchtask.composehttptimeout"
 	HTTP_TIMEOUT                         = "launchtask.httptimeout"
 	COMPOSE_STOP_TIMEOUT                 = "cleanpod.timeout"
+	DEFAULT_COMPOSE_STOP_TIMEOUT         = 10
```
Comment (kmala) [inline config/config.go]: this has to be 10s otherwise viper reads its as nanoseconds

Comment (dpaks) [reply config/config.go]: I replaced the timeout value in SetDefault with "10s" since DEFAULT_COMPOSE_STOP_TIMEOUT is used in multiple places (incl. test file) as '10'

**Review by kmala — APPROVED** [review]
